packages/p2p/src/components/orders/popup.jsx

```diff
@@ -7,153 +7,126 @@ import FormError from '../form/error.jsx';
+                                                    {...field}
+                                                    onChange={() =>
+                                                        setFieldValue('need_confirmation', !values.need_confirmation)
+                                                    }
+                                                    defaultChecked={values.need_confirmation}
+                                                    label={localize('I have received {{amount}} {{currency}}.', {
+                                                        amount: order_information.price_display,
+                                                        currency: order_information.local_currency,
+                                                    })}
+                                                    classNameLabel='orders__popup-field_text'
+                                                />
+                                            )}
+                                        </Field>
+                                    </div>
                                 </div>
                             </ThemedScrollbars>
                             <div className='orders__popup-footer'>
-                                {api_error_message && <FormError message={api_error_message} />}
+                                {status && status.error_message && <FormError message={status.error_message} />}
                                 {has_cancel && (
                                     <Button onClick={onCancel} secondary>
                                         {cancel_text}
                                     </Button>
                                 )}
-                                <Button
-                                    onClick={() => onClickConfirm(this.setApiError)}
-                                    primary
-                                    is_disabled={payment_confirm && disable_confirm}
-                                >
-                                    {confirm_text}
+                                <Button is_disabled={isSubmitting || !values.need_confirmation} primary>
+                                    {localize('Release {{amount}} {{currency}}', {
+                                        amount: order_information.amount_display,
+                                        currency: order_information.account_currency,
+                                    })}
                                 </Button>
                             </div>
-                        </>
+                        </Form>
                     )}
-                </div>
-            </>
-        );
-    }
-}
+                </Formik>
+            ) : (
+                <React.Fragment>
+                    <ThemedScrollbars height='130px'>
+                        <div className='orders__popup-content'>
+                            {message}
+                            {payment_confirm ? (
```

Comment:
```suggestion
                            {payment_confirm && (
```

no need to pass in empty string at line 105?